src/library_wasmfs.js

```diff
@@ -0,0 +1,19 @@
+var WasmfsLibrary = {
+$WASMFS: {
```

Comment:
Let's refactor this to avoid the single global `WASMFS` object. As @sbc100 mentioned that is a pattern that is not optimal because of code size.

Instead, something like

```js
$wasmfsBuffers: [null, [], []],

emscripten_wasmfs_printchar__deps: ['$wasmfsBuffers'],
emscripten_wasmfs_printchar: function(stream, curr) {
  // use wasmfsBuffers[1] etc.
```
That way we define plain C functions for the C++ to call, and those functions can add dependencies on other JS things they need.

In the C++ file we'd declare
```c
int emscripten_wasmfs_printchar(..);
```
and then call that - without knowing it is actually calling into JS.

Note that this also avoids the `EM_ASM` in the C++, which is also good.